core/include/utils/enumerate.hpp

```diff
@@ -6,6 +6,8 @@
  */
 #pragma once
 
+#include "utils/indexing.hpp"
```

Comment:
Is this intended for the iterator index?

Reply:
Ah, i wanted to make a range iterator as a handy tool, well for later.